lib/parsers/pdf_parser/xref_table.rb

```diff
@@ -0,0 +1,99 @@
+class FormatParser::PDFParser::XRefTable
+  include FormatParser::IOUtils
+
+  XRef = Struct.new(:idx, :offset, :generation_number, :entry_type)
+
+  def self.parse(io)
+    new(io).parse
+  end
+
+  def initialize(io)
+    @io = io
```

Comment:
I doubt this object needs to be stateful. It can me a stateless module that we include in the parser itself maybe?